### PR TITLE
Fix a lot of incorrect information

### PR DESCRIPTION
**Pull request by TylerFisher**
Found a couple typos in the data, a bad header name and a malformed date.

### CODE REVIEW

**Comment by gigamonkey** [comment]
FWIW, the fixed date in the NY primary seems to be correct based on:

http://www.elections.ny.gov/NYSBOE/law/2018PoliticalCalendar.pdf

**Comment by TylerFisher** [comment]
Hi, so this is no longer fixing a couple typos. I have found numerous errors in the data, based on the secretary of state websites. I only made my way through about half of the states so far, and I wanted to check in and make sure I'm not crazy.


### Arkansas
arkansas

Arkansas' voter registration was missing, found it [on their election calendar](https://www.sos.arkansas.gov/uploads/elections/2018ElectionCalendar.pdf).

### Colorado
colorado

Colorado's vote by mail application deadline is in fact June 18th.

colorado-2

**Comment by TylerFisher** [comment]
Hi, so this is no longer fixing a couple typos. I have found numerous errors in the data, based on the secretary of state websites. I only made my way through about half of the states so far, and I wanted to check in and make sure I'm not crazy.

Below are screenshots from where I found the information, almost always directly on the secretary of state's website. If there's a more reliable source of information I should be using, let me know.

### Arkansas
arkansas

Arkansas' voter registration was missing, found it [on their election calendar](https://www.sos.arkansas.gov/uploads/elections/2018ElectionCalendar.pdf).

### Colorado
colorado

Colorado's vote by mail application deadline is in fact June 18th.

colorado-2

Early voting runs from June 18th to June 26th. Both of these dates are from [the secretary of state's election calendar](https://www.sos.state.co.us/pubs/elections/calendars/2018ElectionCalendar.pdf).

### Hawaii
hawaii

[The secretary of state calendar](http://elections.hawaii.gov/about-us/calendar/action~agenda/request_format~html/) clearly states the open/close dates for early voting and the vote by mail application deadline.

### Idaho
idaho

[The secretary of state calendar](https://sos.idaho.gov/elect/2018Calendar.html) notes the last day for application of a mail-in ballot as May 4, not May 9.

### Illinois
illinois

[The secretary of state calendar](https://www.elections.il.gov/Downloads/ElectionInformation/PDF/2018ElectionCalendar.pdf) says that early voting ends on March 19, not March 17.

illinois-2

The same calendar says that vote by mail applications are due March 15, not March 19.

### Indiana
indiana
indiana-2

Absentee-in-person voting in Indiana begins 28 days before the election. The election is on May 8, so that would mean April 10 is the beginning of absentee-in-person voting. [The secretary of state's calendar](https://www.in.gov/sos/elections/files/2018%20Calendar%20Brochure.pdf) notes the same date. This calendar marks the beginning of early voting as March 6, and closing on April 6.

### Kansas
kansas

[Kansas's secretary of state site](http://www.kssos.org/elections/elections_registration_voting.html) (which was missing -- I added to the state reference file) says that the last day to register to vote for the primary is July 17

### Maryland
maryland

[Maryland's election calendar](http://www.elections.state.md.us/elections/2018/2018_Election_Calendar.pdf) says that early voting begins on June 14, not June 16.

### Massachusetts
massachusetts

In Massachusetts, [the only information I could find](https://www.sec.state.ma.us/ele/eleifv/howreg.htm) was the general rules about registration, which state that the voter registration deadline is 20 days prior to the election. The election is September 18, so 20 days before that is August 29.

### Minnesota
minnesota

[Minnesota's website](http://www.sos.state.mn.us/elections-voting/other-ways-to-vote/vote-early-by-mail/) claims you can request a vote by mail ballot up until the day before the election.

### Nebraska
nebraska
nebraska-2

In Nebraska, early voting ballots are available 30 days before the election. That's April 16, not April 10. [The 2018 election calendar](http://www.sos.ne.gov/elec/2018/pdf/2018-election-calendar.pdf) also notes this date.

### Nevada
nevada

[Nevada has a handy calendar with all the dates we care about](http://nvsos.gov/sos/elections/2018-election/2018-calendar-events). Voter registration in person closes May 22, not May 14. You can request an absentee ballot by mail until June 7, not June 5.

### New Hampshire
new-hampshire

[New Hampshire's calendar](http://sos.nh.gov/WorkArea/DownloadAsset.aspx?id=8589972718) notes that voter registration closes on August 29.

new-hampshire-2

The only information I could find about vote by mail in New Hampshire was from the [Federal Voting Assistance Program](https://www.fvap.gov/new-hampshire). They claim you can apply for absentee ballots technically until the day before the election.

File: 2018/g2018_vr.csv [hunk 1]
```diff
@@ -4,7 +4,7 @@ AK,2018-10-07,2018-10-07,,,
 AZ,2018-10-09,2018-10-09,,,
 AR,2018-10-07,,,,
 CA,2018-10-22,2018-10-22,,,
-CO,,,,,Voters can register at a polling center up to and including Election Day.
+CO,2018-11-06,,,,Voters can register at a polling center up to and including Election Day.
```
Comment (hodgesmr) [inline 2018/g2018_vr.csv]: There's been some conversation about same-day registration. For example, I had opened a Colorado PR (#4) that was closed. @fpereyra has said:
>we prefer to leave the dates blank for states with no real voter registration deadlines and instead include a note to further explain

I've also opened an Issue (#31) asking how to handle states that have no real in-person deadline but do have online deadlines.

File: 2018/g2018_vr.csv [hunk 2]
```diff
@@ -4,31 +4,31 @@ AK,2018-10-07,2018-10-07,,,
 AZ,2018-10-09,2018-10-09,,,
 AR,2018-10-07,,,,
 CA,2018-10-22,2018-10-22,,,
-CO,,,,,Voters can register at a polling center up to and including Election Day.
-CT,2018-10-07,2018-10-07,,,
+CO,2018-11-06,,,,Voters can register at a polling center up to and including Election Day.
```
Comment (hodgesmr) [inline 2018/g2018_vr.csv]: There's been some conversation about same-day registration. For example, I had opened a Colorado PR with this change (#4), that was closed. @fpereyra has said:
>we prefer to leave the dates blank for states with no real voter registration deadlines and instead include a note to further explain

I've also opened an Issue (#31) asking how to handle states that have no real in-person deadline but do have online deadlines.

File: 2018/g2018_vr.csv [hunk 3]
```diff
@@ -4,31 +4,31 @@ AK,2018-10-07,2018-10-07,,,
...
+MN,2018-10-16,2018-10-16,,,"The deadline to register online is 11:59 p.m., 21 days before Election Day. The deadline to register on paper is 5 p.m., 21 days before Election Day. Otherwise, you can register on Election Day at your polling place."
 MS,2018-10-08,,,,10/9/2018 is the deadline to register to vote by mail
 MO,2018-10-10,2018-10-10,,,
 MT,2018-10-09,,,,"Regular voter registration closes at 5 p.m. on 10/9/2018.
 Individuals can late-register at the county election office beginning the next day and through close of polls on election day."
 NE,2018-10-19,2018-10-19,,,
-NV,2018-10-09,2018-10-18,,,In-person deadline is 10/16/2018; online voter registration deadline is 10/18/2018
-NH,,,,,"GENERAL ELECTION DAY- November 6, 2018 - unregistered voters may register and vote on this day."
+NV,2018-10-16,2018-10-18,,,In-person deadline is 10/16/2018; online voter registration deadline is 10/18/2018
+NH,2018-11-06,,,,"GENERAL ELECTION DAY- November 6, 2018 - unregistered voters may register and vote on this day."
```
Comment (hodgesmr) [inline 2018/g2018_vr.csv]: See the [other comment](https://github.com/democrats/election-calendar/pull/36/files#r170463780) and #7

File: 2018/g2018_vr.csv [hunk 4]
```diff
@@ -37,19 +37,19 @@ ND,,,,,Voters do not register in North Dakota
...
+PA,2018-10-09,2018-10-07,,,
 RI,2018-10-07,2018-10-07,,,
 SC,2018-10-07,2018-10-07,,,
 SD,2018-10-22,,,,
-TN,2018-10-07,2018-10-07,,,
+TN,2018-10-09,2018-10-07,,,
 TX,2018-10-07,,,,
 UT,2018-10-30,2018-10-30,,,
-VT,,,,,"Voters in Vermont can register to vote on any day, including Election Day"
+VT,2018-11-06,,,,"Voters in Vermont can register to vote on any day, including Election Day"
```
Comment (hodgesmr) [inline 2018/g2018_vr.csv]: See the [other comment](https://github.com/democrats/election-calendar/pull/36/files#r170463780) and #10

File: 2018/g2018_vr.csv [hunk 5]
```diff
@@ -37,19 +37,19 @@ ND,,,,,Voters do not register in North Dakota
...
 VA,2018-10-15,2018-10-15,,,
-WA,2018-10-08,2018-10-08,,,
+WA,2018-10-29,2018-10-08,,,
 WV,2018-10-16,2018-10-16,,,
-WI,,,,,"Online: up to 20 days before the election. At the polling place on Election Day: you may register at the polls on Election Day."
-WY,,,,,"Register at the polls on Election Day"
+WI,2018-11-06,2018-10-17,,,"Online: up to 20 days before the election. At the polling place on Election Day: you may register at the polls on Election Day."
```
Comment (hodgesmr) [inline 2018/g2018_vr.csv]: See the [other comment](https://github.com/democrats/election-calendar/pull/36/files#r170463780) and #8

File: 2018/p2018_federal_ev.csv [hunk 6]
```diff
@@ -1,50 +1,50 @@
...
-RI,,,2018-08-21,2018-09-11,
-SC,,,2018-06-08,2018-06-11,
-SD,2018-04-20,2018-06-04,2018-06-01,2018-06-05,In-person absentee voting; varies by county
-TN,2018-07-13,2018-07-27,2018-07-26,2018-08-02,
-TX,2018-02-19,2018-03-02,2018-02-23,2018-03-06,
+RI,,,2018-08-22,2018-09-11,
+SC,2018-05-14,2018-06-11,2018-06-08,2018-06-11,
+SD,2018-04-20,2018-06-04,2018-06-04,2018-06-05,In-person absentee voting; varies by county
+TN,2018-07-13,2018-07-28,2018-07-26,2018-08-02,
+TX,2018-02-20,2018-03-02,2018-02-23,2018-03-06,
```
Comment (hodgesmr) [inline 2018/p2018_federal_ev.csv]: This was fixed in #41, which has been merged.

**Comment by fpereyra** [comment]
thanks all! per @hodgesmr, we're leaving VR deadlines blank in states where folks can register and vote on election day (we'll make this more explicit in the docs too).  big thanks to @TylerFisher for corrections with screenshots!

**Comment by TylerFisher** [comment]
Thank you for this much needed resource!

**Pull request merged**